modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/beta_indicator/BetaButton.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import ClayPopover, {ALIGN_POSITIONS} from '@clayui/popover';
+import {useId} from '@clayui/shared';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import {sub} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import LearnMessage, {
+	LearnResourcesContext,
+} from '../learn_message/LearnMessage';
+import BetaBadge, {betaClassNames} from './BetaBadge';
+
+export default function BetaButton({
+	learnResourceContext,
+	tooltipAlign = 'top',
+}: {
+	learnResourceContext: object;
+	tooltipAlign: typeof ALIGN_POSITIONS[number];
+}) {
+	const ariaControlsId = useId();
+
+	const [show, setShow] = useState(false);
+
+	const [learnMessageNode, setLearnMessageNode] = useState(
+		document.createElement('span')
+	);
+
+	useEffect(() => {
+		// eslint-disable-next-line @liferay/portal/no-react-dom-render
+		ReactDOM.render(
```

Comment:
Out of curiosity, what makes this necessary?

Reply:
This kinda looks weird to me, too. I've never seen a component being rendered inside a `useEffect` hook before.

Reply:
Thanks for pointing this out. It is for sure, indeed that's the part I'm most concerned with by far. I'd be totally fine if we come up with a different solution.

Here is the rationale. In order to achieve this:
![image](https://github.com/liferay-frontend/liferay-portal/assets/861014/e5fc9902-b489-4856-b714-d69b30a019a6)
we need to interpolate a translation key `see-x-for-more-information` with a link. Such a link (both the text and the target URL) is given by the `<LearnMessage>` react component output. 

There are a couple of cases where DXP is using `{ReactDOMServer.renderToString()` to get the markup of a component as string. I tried to avoid this by using an auxiliar dom node to render.

A much better solution is to not obey the designs and use a full link in the <LearnMessage>. This would avoid browser-side interpolation of a translation key, yielding the following result:
![image](https://github.com/liferay-frontend/liferay-portal/assets/861014/45864aa8-410c-46ea-922c-0db5d359b97c)

In code this would be much simpler too:
```
	return (
		<LearnResourcesContext.Provider value={learnResourceContext}>
			<ClayTooltipProvider>
				...
					<ClayPopover
...						trigger={
...						}
					>
						{Liferay.Language.get('this-feature-is-in-testing')}

                                                // no more lang key interpolation, all the text in the link comes from learn message
						<LearnMessage
							resource="frontend-js-components-web"
							resourceKey="beta-features-page"
						/>
					</ClayPopover>
				</div>
			</ClayTooltipProvider>
		</LearnResourcesContext.Provider>
	);
```

In addition, the learn JSON would include the full text:

```
{
	"beta-features-page": {
		"en_US": {
			"message": "See beta features page for more information",
//			"message": "beta features page",
			"url": "https://learn.liferay.com/w/dxp/system-administration/configuring-liferay/feature-flags"
		}
	}
}
``` 

So if we're fine with this deviation from designs, I'd happily go this way.

Another alternative is to pass the link contents from the backend, so there'd be no need to use `<LearnMessage>` in react side. This way we ccould still obey the design at the cost of not reusing the standard learn markup

Thoughts?

Reply:
IMO the cost of conforming to the design here isn't worth the results. It's just some blue words, no one is gonna notice I promise 😆